### PR TITLE
Support the use of client certificates

### PR DESCRIPTION
**Pull request by cristian-radu**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,21 +26,37 @@ Note: docker registry must be [v2](https://docs.docker.com/registry/spec/api/).
 
 * `insecure_registries`: *Optional.* An array of CIDRs or `host:port` addresses
   to whitelist for insecure access (either `http` or unverified `https`).
-  This option overrides any entries in `ca_certs` with the same address.
+  This option overrides any entries in `certificates` with the same address.
 
 * `registry_mirror`: *Optional.* A URL pointing to a docker registry mirror service.
 
-* `ca_certs`: *Optional.* An array of objects with the following format:
+* `certificates`: *Optional.* An array of objects with the following format:
```
Comment (vito) [inline README.md]: Hmm, I think this is less descriptive. Does it make sense to group CA certs and client certs together? They feel like they're for different things (client verifying the server vs server verifying the client) and should be configured separately.

**Comment by cristian-radu** [comment]
Hi, I have changed it following your review comments. It will treat client certificates separately from ca certificates now.

File: cmd/check/main.go [hunk 2]
```diff
@@ -180,10 +180,20 @@ func makeTransport(logger lager.Logger, request CheckRequest, registryHost strin
...
+	if len(request.Source.ClientCerts) > 0 {
+		baseTransport.TLSClientConfig = &tls.Config{
+			RootCAs: caCertPool,
+			Certificates: setClientCert(registryHost, request.Source.ClientCerts),
+		}
+	}
+
 	authTransport := transport.NewTransport(baseTransport)
 
+	timeout := time.Duration(10 * time.Second)
```
Comment (vito) [inline cmd/check/main.go]: I don't think we should sneak this change in. Separate PR? Why was this needed?

File: cmd/check/main.go [hunk 3]
```diff
@@ -302,3 +312,14 @@ func retryRoundTripper(logger lager.Logger, rt http.RoundTripper) http.RoundTrip
 		RoundTripper: rt,
 	}
 }
+
+func setClientCert(registry string, list []ClientCertKey) []tls.Certificate {
+	  var clientCert []tls.Certificate
```
Comment (vito) [inline cmd/check/main.go]: Weird spacing here; mind running `go fmt ./...`?

File: cmd/check/main.go (hunk 2 above)
Comment (cristian-radu) [inline cmd/check/main.go]: OK, I will move it to a separate PR.

File: cmd/check/main.go (hunk 3 above)
Comment (cristian-radu) [inline cmd/check/main.go]: OK, I ran fmt on it.

**Comment by cristian-radu** [comment]
@vito I have made the changes you requested. Please take a look.

File: cmd/check/main.go [hunk 4]
```diff
@@ -302,3 +309,14 @@ func retryRoundTripper(logger lager.Logger, rt http.RoundTripper) http.RoundTrip
 		RoundTripper: rt,
 	}
 }
+
+func setClientCert(registry string, list []ClientCertKey) []tls.Certificate {
+	var clientCert []tls.Certificate
+	for _, r := range list {
+		if r.Domain == registry {
+			certKey, _ := tls.X509KeyPair([]byte(r.Cert), []byte(r.Key))
```
Comment (vito) [inline cmd/check/main.go]: Probably shouldn't drop this error - I could see someone easily messing up the key or cert.

Comment (cristian-radu) [inline cmd/check/main.go]: Good point. I have updated the function.

**Comment by pivotal-issuemaster** [comment]
@cristian-radu Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=concourse/docker-image-resource&pullRequestId=74) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@cristian-radu Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=concourse/docker-image-resource&pullRequestId=74)!

**Comment by vito** [comment]
looks great, thanks!
